cmd/minikube/cmd/config/config.go

```diff
@@ -46,122 +49,145 @@ type Setting struct {
 var settings = []Setting{
 	{
 		name:          "driver",
+		description:   "Used to specify the driver to run Kubernetes in. The list of available drivers depends on operating system.",
 		set:           SetString,
 		validDefaults: driver.SupportedDrivers,
 		validations:   []setFn{IsValidDriver},
 		callbacks:     []setFn{RequiresRestartMsg},
 	},
 	{
 		name:        "vm-driver",
+		description: "DEPRECATED, use driver instead.",
 		set:         SetString,
 		validations: []setFn{IsValidDriver},
 		callbacks:   []setFn{RequiresRestartMsg},
 	},
 	{
 		name:        "container-runtime",
+		description: "The container runtime to be used (docker, cri-o, containerd). (default \"docker\")",
 		set:         SetString,
 		validations: []setFn{IsValidRuntime},
 		callbacks:   []setFn{RequiresRestartMsg},
 	},
 	{
-		name:      "feature-gates",
-		set:       SetString,
-		callbacks: []setFn{RequiresRestartMsg},
+		name:        "feature-gates",
+		description: "A set of key=value pairs that describe feature gates for alpha/experimental features.",
+		set:         SetString,
+		callbacks:   []setFn{RequiresRestartMsg},
 	},
 	{
 		name:        "v",
+		description: "number for the log level verbosity",
 		set:         SetInt,
 		validations: []setFn{IsPositive},
 	},
 	{
 		name:        "cpus",
+		description: "Number of CPUs allocated to each minikube node",
 		set:         SetInt,
 		validations: []setFn{IsValidCPUs},
 		callbacks:   []setFn{RequiresRestartMsg},
 	},
 	{
 		name:        "disk-size",
+		description: "Disk size allocated to the minikube VM (format: <number>[<unit>], where unit = b, k, m or g). (default \"20000mb\")",
 		set:         SetString,
 		validations: []setFn{IsValidDiskSize},
 		callbacks:   []setFn{RequiresRestartMsg},
 	},
 	{
 		name:        "host-only-cidr",
+		description: "The CIDR to be used for the minikube VM (virtualbox driver only) (default \"192.168.99.1/24\")",
 		set:         SetString,
 		validations: []setFn{IsValidCIDR},
 	},
 	{
 		name:        "memory",
+		description: "Amount of RAM to allocate to Kubernetes (format: <number>[<unit>], where unit = b, k, m or g).",
 		set:         SetString,
 		validations: []setFn{IsValidMemory},
 		callbacks:   []setFn{RequiresRestartMsg},
 	},
 	{
 		name:        "log_dir",
+		description: "If non-empty, write log files in this directory",
 		set:         SetString,
 		validations: []setFn{IsValidPath},
 	},
 	{
-		name: "kubernetes-version",
-		set:  SetString,
+		name:        "kubernetes-version",
+		description: "The Kubernetes version that the minikube VM will use (ex: v1.2.3, 'stable' for v1.20.7, 'latest' for v1.22.0-alpha.2). Defaults to 'stable'.",
```

Comment:
The stable and latest version are constantly changing, we should remove these versions as they're going to get outdated quickly.